Assets/_Scripts/GameState.cs

```diff
@@ -0,0 +1,40 @@
+
+using UnityEngine;
+using System.Collections.Generic;
+
+// This class stores the current game state, not specific to any particular scene.
+// Might not need to be MonoBehaviour, but I'll leave it open for now
+[RequireComponent (typeof(ProgressData))]
+public class GameState : MonoBehaviour {
+    public ProgressData progressData;
+
+    public List<StageInfoContainer> stages {get; set;}
+
+    public void Initialize() {
+        this.InitializeStages();
+    }
+
+  
+    public StageInfoContainer GetCurrentStage() {
+        return this.stages[this.progressData.currentStageIndex];
+    }
+
+    public void SetStageIndex(int index) {
+        this.progressData.currentStageIndex = index;
+    }
+
+    public void SetNextStageIndex(int index) {
+        this.progressData.nextStageIndex = index;
+    }
+
+    private void InitializeStages() {
```

Comment:
Also a good idea to avoid generation logic in a method with the name `.*State`. Should realistically only need to provide basic Getter&Setter methods, but GameState is small so its not a big deal for now.

Reply:
Usually the common design pattern is to use either a `Generator` or `Factory` class which produces these initial states.

Reply:
This can be done later, but I don't think it's necessary right now. You can design it if you want.